include/storage/shared_datatype.hpp

```diff
@@ -115,9 +92,59 @@ class DataLayout
         }
     }
 
+    virtual inline void *GetBlockPtr(char *shared_memory, const std::string &name) const = 0;
```

Comment:
Why a `char*`? The semantics are not clear here.

Reply:
`char *` is old code - I've moved everything to `void *` which is more appropriate.